### ISSUE TITLE
Allow disabling of refresh

### ISSUE THREAD

**Issue by waynebrantley** [Azure/AppConfiguration-DotnetProvider#403]
Currently the app config always refreshes and defaults to 30 seconds.  Examples of when this is desired are setting up app settings in UAT (and using the free SKU) or when doing a [push refresh](https://learn.microsoft.com/en-us/azure/azure-app-configuration/enable-dynamic-configuration-dotnet-core-push-refresh)

Currently, the only way to change this is to add a new value and set a new expiration time.

```c#
 IConfiguration configuration = new ConfigurationBuilder()
                .AddAzureAppConfiguration(options =>
                {
                    options.Connect(appConfigurationConnectionString);
                    options.ConfigureRefresh(refresh =>
                        refresh
                            .Register("TestApp:Settings:Message")
                            .SetCacheExpiration(TimeSpan.FromDays(1))  // Important: Reduce poll frequency
                    );
                }).Build();
```

This actually causes another request for the "TestApp:Settings:Message" value - even though that initial value was already loaded in the normal config load.  This initial request is not necessary most of the time.  Depending on the 'select' that was used, the key may not have been loaded - so the easy thing to do was to just load it.   However, I think you could check the current config for a value for it and only if no value do you try and load it. 

Anyway, would be nice to just disable refresh.  (Honestly that would probably have been a better default than the 30 second refresh all keys way).   So something like:

```c#
 IConfiguration configuration = new ConfigurationBuilder()
                .AddAzureAppConfiguration(options =>
                {
                    options.Connect(appConfigurationConnectionString);
                    options.ConfigureRefresh(refresh =>
                        refresh.Disable()
                    );
                }).Build();
```
Or the like I mentioned the default (Which is a breaking change, but surely most people use sentinel value already) should be disabled.  So that means the 'ConfigureRefresh' call is what enables it.  So the new config to disable a refresh would simply be nothing:

```c#
 IConfiguration configuration = new ConfigurationBuilder()
                .AddAzureAppConfiguration(options =>
                {
                    options.Connect(appConfigurationConnectionString);
                }).Build();
```

Finally, concerning the 'automatic load of the sentinel value' we should have something that causes it to use the value already loaded. 
 I think most of the time a .Select("TestApp:*") is used - so it is likely already loaded.  Example api enhancement:

```c#
 IConfiguration configuration = new ConfigurationBuilder()
                .AddAzureAppConfiguration(options =>
                {
                    options.Connect(appConfigurationConnectionString);
                    options.ConfigureRefresh(refresh =>
                        refresh.InitialValueFromConfiguration()
                    );
                }).Build();
```

So, in summary:
- Make it simple to turn off refresh of application configuration
- Reduce the api calls if refreshing by using the pre-loaded value

Thanks for considering

**Comment by waynebrantley** [comment]
@amerjusupovic would this be something you would consider doing or accepting a PR on?

**Comment by waynebrantley** [comment]
@amerjusupovic I can certainly be waaaayyy off here, but the documentation seemed to suggest that the default refresh was 30 seconds for all keys.   To that end they have articles on how to use a sentinel key to reduce those requests.   

https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.azureappconfiguration.azureappconfigurationrefreshoptions.setcacheexpiration?view=azure-dotnet

https://learn.microsoft.com/en-us/azure/azure-app-configuration/enable-dynamic-configuration-aspnet-core?tabs=core6x#add-a-sentinel-key

However, if I am totally off base and it does not refresh at all by default - then the first point is n/a.?

**Comment by amerjusupovic** [comment]
@waynebrantley These articles are referring to key-values that are registered for refresh, which are different from key values that are loaded only by calling `AzureAppConfigurationOptions.Select()`. If a key-value like "Sentinel" is registered using `refreshOptions.Register("Sentinel", refreshAll: true)` like in the article you linked, then the entire configuration may refresh if the cache expires or "Sentinel" changes in the App Configuration store. You also have to trigger a refresh somehow, either through the poll or push model, to actually refresh the configuration. 

As you mentioned in your proposal, you do have to call `ConfigureRefresh()` to enable refresh and it isn't enabled by default.
